### DISCUSSION TITLE
Should reading the body be buffered?

### DISCUSSION THREAD

**Discussion by kathampy** [sagebind/isahc#102] [Questions] [answered]
When reading the body with `read_to_end()` or `take()`, is it necessary to use a BufReader? Since it's wrapping libcurl, is sufficient buffering already being performed internally?

**Comment by sagebind** [comment, marked as answer]
No, it is not necessary; libcurl uses its own buffering mechanism already (I think the default buffer size is 16K, though it may pick a different size depending on other factors).

**Comment by sagebind** [comment]
I'll add a brief note about this in the documentation.
